### ISSUE TITLE
git hook server stops working after some time

### ISSUE THREAD

**Issue by jonorthwash** [goavki/phenny#258]
It seems that any time a packet is dealt with poorly by the git server, the module crashes and it becomes unable to deal with further requests.

Here's an example of a recent one that I can fix.

```
----------------------------------------
Exception happened during processing of request from ('192.30.252.42', 53557)
Traceback (most recent call last):
  File "/usr/lib/python3.5/socketserver.py", line 313, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 341, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python3.5/socketserver.py", line 354, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.5/socketserver.py", line 681, in __init__
    self.handle()
  File "/usr/lib/python3.5/http/server.py", line 422, in handle
    self.handle_one_request()
  File "/usr/lib/python3.5/http/server.py", line 410, in handle_one_request
    method()
  File "/home/begiak/phenny/modules/git.py", line 160, in do_POST
    opt = 'to ' + data['issue']['assignee']
TypeError: Can't convert 'dict' object to str implicitly
----------------------------------------
```

I think we just need better error handling.

**Comment by jonorthwash** [comment]
This particular example was handled by https://github.com/goavki/phenny/commit/d5b046

But we can't always keep up 100% with changes in the payloads, so we need a better error handling approach in general.

**Comment by jonorthwash** [comment]
Duplicate of #133.

**Comment by shardulc** [comment]
Closed via #279.
